### PR TITLE
Dsl update2

### PR DESCRIPTION
**Pull request by praveenraj2018**
Below updates are included in this PR:

1) Update nf-core modules to the latest available
2) Update code for the latest DSL2 syntax changes.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/rnavar/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/rnavar _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/prepare_genome.nf [hunk 1]
```diff
@@ -93,17 +98,20 @@ workflow PREPARE_GENOME {
...
-        if((!ch_star_index) || getIndexVersion(ch_star_index) != '2.7.4a'){
-            ch_star_index   = STAR_GENOMEGENERATE(ch_fasta,ch_gtf).index
-            ch_versions     = ch_versions.mix(STAR_GENOMEGENERATE.out.versions)
-        }
+        //print(getIndexVersion(ch_star_index))
+        //ch_star_index.view()
+        //if((!ch_star_index) || getIndexVersion(ch_star_index) != '2.7.4a'){
+        //    ch_star_index   = STAR_GENOMEGENERATE(ch_fasta,ch_gtf).index
+        //    ch_versions     = ch_versions.mix(STAR_GENOMEGENERATE.out.versions)
+        //}
```
Comment (maxulysse) [inline subworkflows/local/prepare_genome.nf:114]: still no luck with this one?

**Review by maxulysse — APPROVED** [review]

File: subworkflows/local/prepare_genome.nf (hunk 1 above)
Comment (maxulysse) [reply subworkflows/local/prepare_genome.nf:114]: I think we can try to fix it in another PR, and merge this one so that we can go on with other things

**Comment by praveenraj2018** [comment]
Reg. YAML lint errors - I could see many of them are from snpeff/meta.yml. I fixed the rest that are from branch.yml, etc...
```
praveen@praveen-VirtualBox:~/coderepo/rnavar$ yamllint $(find . -type f -name "*.yml" -o -name "*.yaml") -c ./.yamllint.yml
./nf-core/modules/snpeff/meta.yml
  7:9       error    wrong indentation: expected 10 but found 8  (indentation)
  15:9      error    wrong indentation: expected 10 but found 8  (indentation)
  21:9      error    wrong indentation: expected 10 but found 8  (indentation)
  27:9      error    wrong indentation: expected 10 but found 8  (indentation)
  32:9      error    wrong indentation: expected 10 but found 8  (indentation)
  36:9      error    wrong indentation: expected 10 but found 8  (indentation)
  40:9      error    wrong indentation: expected 10 but found 8  (indentation)
  45:9      error    wrong indentation: expected 10 but found 8  (indentation)
  50:9      error    wrong indentation: expected 10 but found 8  (indentation)
  54:9      error    wrong indentation: expected 10 but found 8  (indentation)
```
How do we fix the snpeff lint errros? Should it be changed in the modules and do a PR? How did it work for you in Sarek?

**Comment by maxulysse** [comment]
I'll make a PR in the modules, thanks

**Comment by praveenraj2018** [comment]
How do we fix the YAML lint. It is coming out from 

> I'll make a PR in the modules, thanks

Thanks. Then I should be able to pass the tests and merge.

**Comment by ewels** [comment]
The old markdownlint and yamllint CI checks are on `pull_request` and are coming from the existing code base. They will stop running _after_ this PR is merged.

I don't think it matters if those fail, as long as the Prettier and editoconfig ones are passing 👍🏻

**Comment by ewels** [comment]
Note that commits eef058f and 06e1c81 should be part of this PR if the aim is to update the template (see https://github.com/nf-core/rnavar/pull/27). The exact commits need to be part of the git history on `dev` for the template sync mechanism to work nicely.

x-ref [discussion on Slack](https://nfcore.slack.com/archives/CE5LG7WMB/p1648119402159369).

File: .yamllint.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+extends: default
```
Comment (ewels) [inline .yamllint.yml:1]: This file needs to be deleted.

Comment (ewels) [reply .yamllint.yml:1]: You're also missing the `.editorconfig.yml` file that needs to be here.

**Comment by praveenraj2018** [comment]
Not needed

**Pull request closed without merging**